src/leiningen/protoc.clj

```diff
@@ -193,9 +193,12 @@
 
 (defn get-arch
   []
-  (if-let [arch (leiningen.core.utils/get-arch)]
-    (name (if (= arch :x86) :x86_32 arch))
-    (throw (Exception. "Leiningen failed to detect the processor architecture"))))
+  ;; backport fix from leiningen: https://github.com/technomancy/leiningen/commit/60f7ab7cb132ad5e495688ec4f066a575c9996c9
```

Comment:
is this because a new version of lein has not been released? i think we have been asking M1 folk to build their own version of lein instead

Reply:
> is this because a new version of lein has not been released?

correct 

> i think we have been asking M1 folk to build their own version of lein instead

this is also possible, you can use the current stable lein version to build a current lein and add that to your $PATH -- this PR removes the friction of doing that though

We can also document the 'build your own lein' in the setup docs if we decide we don't want to bump the version here

Reply:
is this backport sufficient though? can someone otherwise use an unpatched version of lein on an M1 laptop?

Reply:
> can someone otherwise use an unpatched version of lein on an M1 laptop?

yes they can, and for lein-protoc a manual lein install from the git repo works just fine (side note: I could not get a `brew install leiningen --head` to work)

I think the need for this PR goes away then since a new release resolves this and we have a followup to document "how to m1"